### PR TITLE
Feature/pet image view pager

### LINKED ISSUE #46: Add a image pager to FragmentDetails for animals with multiple pictures
The detail fragment current shows one image on the animal.  Replace the imageview with a view pager or image switcher or some UI widget that lets you swipe left and right to see more pictures.

If the user presses on a single picture, they are brought to a full screen view of the pet image and they can continue to scroll in this full-screen view.

### PR DESCRIPTION
**Pull request by JohnLeeroy**
Fixes #46 

Here's a gif of it in action.
petscroll

### CODE REVIEW

File: app/src/main/java/com/codefororlando/petadoption/recyclerview/HorizontalViewPagerIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+class HorizontalViewPagerIndicator : LinearLayout, ViewPager.OnPageChangeListener, ViewPager.OnAdapterChangeListener {
+
+    private var pageCount = 0
+    private var selectedIndex = 0
+    private var indicatorViews: MutableList<View> = mutableListOf()
+    private lateinit var scaleUpAnim: ScaleAnimation
+    private lateinit var scaleDownAnim: ScaleAnimation
+
+    constructor(context: Context) : super(context) {
```
Comment (ToxicBakery) [inline app/src/main/java/com/codefororlando/petadoption/recyclerview/HorizontalViewPagerIndicator.kt]: You don't need the repeating constructors if you use JvmOverloads.

```kotlin
class HorizontalViewPagerIndicator @JvmOverloads constructor(
   context: Context,
   attrs: AttributeSet? = null,
   defStyleAttr: Int = 0
) : LinearLayout, ViewPager.OnPageChangeListener, ViewPager.OnAdapterChangeListener {
```

Then remove the unnecessary `init()` function and use `init{}` then change your `lateinit` vars to `val`s

File: app/src/main/java/com/codefororlando/petadoption/view/PetImageViewPagerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.codefororlando.petadoption.R;
+import com.squareup.picasso.Picasso;
+
+import java.util.List;
+
+class PetImageViewPagerAdapter extends PagerAdapter {
```
Comment (ToxicBakery) [inline app/src/main/java/com/codefororlando/petadoption/view/PetImageViewPagerAdapter.java]: You added a Java class??? :cry:

File: app/src/main/java/com/codefororlando/petadoption/recyclerview/HorizontalViewPagerIndicator.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import android.view.animation.Animation
+import android.view.animation.ScaleAnimation
+import android.widget.LinearLayout
+import com.codefororlando.petadoption.R
+
+class HorizontalViewPagerIndicator : LinearLayout, ViewPager.OnPageChangeListener, ViewPager.OnAdapterChangeListener {
+
+    private var pageCount = 0
+    private var selectedIndex = 0
+    private var indicatorViews: MutableList<View> = mutableListOf()
```
Comment (ToxicBakery) [inline app/src/main/java/com/codefororlando/petadoption/recyclerview/HorizontalViewPagerIndicator.kt]: This should be a `val`

File: app/src/main/java/com/codefororlando/petadoption/recyclerview/HorizontalViewPagerIndicator.kt (hunk 1 above)
Comment (JohnLeeroy) [reply app/src/main/java/com/codefororlando/petadoption/recyclerview/HorizontalViewPagerIndicator.kt]: Thanks for the tip.  Updated!

**Comment by JohnLeeroy** [comment]
Addressed comments by @ToxicBakery , got a thumbs up from @RFonzi. Going to go ahead and merge.

**Pull request merged**